os/arch/arm/src/armv8-m/up_blocktask.c

```diff
@@ -189,6 +189,12 @@ void up_block_task(struct tcb_s *tcb, tstate_t task_state)
 #endif
 #ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION
 				up_mpu_set_register(rtcb->stack_mpu_regs);
+#endif
+#ifdef CONFIG_REG_STACK_OVERFLOW_PROTECTION
+				static int32_t result;
```

Comment:
A better way to use the limit register is to intialize the limit register in the up_initialstate() API and store / restore the register in the up_exception.S file. Then, we will not need to add this code in multiple files.